server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -773,24 +773,11 @@ public Iterator<Setting<?>> settings() {
     public static final Setting<Boolean> USE_DOC_VALUES_SKIPPER = Setting.boolSetting("index.mapping.use_doc_values_skipper", s -> {
         IndexVersion iv = SETTING_INDEX_VERSION_CREATED.get(s);
         if (MODE.get(s) == IndexMode.TIME_SERIES) {
-            if (DiscoveryNode.isStateless(s)) {
```

Comment:
Question: I was looking into this issue and noticed that the `DiscoveryNode.isStateless(s)` calls in the default function might never have worked because s is index settings but stateless.enabled is a **NodeScope** setting. Is that the root cause here?

Also, do we need any stateless-specific handling elsewhere (like in the IndexSettings constructor using nodeSettings)? I'm thinking of the pattern we use for `recoverySourceSyntheticEnabled`. Just want to make sure we're not missing something for the timestamp case.